### DISCUSSION TITLE
How to use CFG++?

### DISCUSSION THREAD

**Discussion by not-ski** [comfyanonymous/ComfyUI#3844] [Q&A]
I saw that "SamplerEulerCFG++" node was added in commit https://github.com/comfyanonymous/ComfyUI/commit/73ca7800192f6bc2e2194966b316b7cc3028e09a, and I'm not sure how to use it.

I took a look at the original repo for CFG++, and it suggests using CFG++ in combination with a CFG of 0.6. I tried this, and I wasn't getting good results at all.

Anyone have any better idea of how best to use this? Thanks!

**Comment by comfyanonymous** [comment]
Try cfg 1.6, The ComfyUI cfg scale actually starts at 1.0 but it starts at 0.0 in some codebases.

**Reply by not-ski** [reply]
Thanks! I actually discovered that CFG 0.6 works fine, but the step count has to be quite high (>=50). This tracks with the CFG++ documentation.

Is there a specific scheduler you recommend using with this? The CFG++ example code appears to use "DDIMScheduler", which I would guess is closest to ddim_uniform in Comfy?

**Comment by brbbbq** [comment]
Here's a video outlining some of the new samplers, including Euler CFG++:
https://youtu.be/-GXJDz8i-Wo

And here's the project page for CFG++:
https://cfgpp-diffusion.github.io/
